File: src/main/java/com/pki/example/dto/IssuerCertificateDTO.java
package com.pki.example.dto;

import com.pki.example.model.CertificateType;
import lombok.Data;
import org.bouncycastle.asn1.x500.X500Name;

import java.util.Date;
@Data
public class IssuerCertificateDTO {

    // Podaci potrebni za X500Name
    private String commonName;
    private String surname;
    private String givenName;
    private String organization;
    private String organizationalUnit;
    private String country;
    private String email;
    private String ownerEmail;

    private String type;

    private Date validFrom;
    private Date validTo;

    //serijski broj izdavaoca, ako je root onda je prazan string
    private String issuerSerialNumber;
}

File: security_frontend/security_app/src/app/model/issuerCertificateDto.ts
export interface IssueCertificateDTO {
  commonName: string;
  surname: string;
  givenName: string;
  organization: string;
  organizationalUnit: string;
  country: string;
  email: string;
  uid?: string; 
  validFrom: Date;
  validTo: Date;
  type: string; 
  ownerEmail: string;
  issuerSerialNumber?: string;
  templateId?: number;        
  subjectAlternativeNames?: string[];
  keyUsage?: string[];
  extendedKeyUsage?: string[];
}